File: Frontend/T-Shirt/src/services/recommendationService.ts
import { api } from "./api";
import type { FrequentlyBoughtResponse } from "../types/recommendation";
import type { Product } from "../types/product";

export const recommendationService = {
  getFrequentlyBoughtTogether: async (productId: number): Promise<FrequentlyBoughtResponse> => {
    const response = await api.get(`/recommendations/frequently-bought/${productId}`);
    return response.data;
  },

  getPersonalizedRecommendations: async (limit = 6): Promise<Product[]> => {
    const response = await api.get("/recommendations/personalized", { params: { limit } });
    return response.data;
  },
};

File: Backend/tshirt-store-backend/src/main/java/com/tshirtstore/service/RecommendationService.java
package com.tshirtstore.service;

import com.tshirtstore.dto.ProductResponse;
import com.tshirtstore.dto.recommendation.FrequentlyBoughtResponse;
import java.util.List;

public interface RecommendationService {
    FrequentlyBoughtResponse getFrequentlyBoughtTogether(Long productId);
    List<ProductResponse> getPersonalizedRecommendations(int limit);
}
